pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -187,6 +188,31 @@ func remaining(t time.Time) time.Duration {
 	return t.Sub(time.Now()).Round(time.Second)
 }
 
+func dumpClusterAutoscalerLogs(client runtimeclient.Client, restClient *rest.RESTClient) {
+	pods := corev1.PodList{}
+	caLabels := map[string]string{
+		"app": "cluster-autoscaler",
+	}
+	if err := client.List(context.TODO(), &pods, []runtimeclient.ListOptionFunc{runtimeclient.MatchingLabels(caLabels)}...); err != nil {
+		glog.Errorf("Error querying api for clusterAutoscaler pod object: %v", err)
+	}
+	if len(pods.Items) > 1 {
+		glog.Errorf("Got more than one autoscaler pods, which is unexpected!!!")
+	}
+	// We're only expecting one pod but let's log from all that
+	// are found. If we see more than one that's indicative of
+	// some unexpected problem and we may as well dump its logs.
+	for _, pod := range pods.Items {
+		req := restClient.Get().Namespace(e2e.TestContext.MachineApiNamespace).Resource("pods").Name(pod.Name).SubResource("log")
+		res := req.Do()
+		raw, err := res.Raw()
+		if err != nil {
+			glog.Errorf("Unable to get pod logs: %v", err)
+		}
+		glog.Infof("\n\n Pod %q logs:\n\n%v", pod.Name, string(raw))
```

Comment:
To capture how many Pods were found consider logging as:

```go
glog.Infof("Dumping pod logs: %d/%d, logs:\n%v", i, len(pods), pod.Name, string(raw))
```

You could then dispense with the conditional:

```go
if len(pods.Items) > 1 {
	glog.Errorf("Got more than one autoscaler pods, which is unexpected!!!")
}
```